crypto/dsa/dsa_asn1.c

```diff
@@ -32,6 +32,15 @@ void DSA_SIG_get0(BIGNUM **pr, BIGNUM **ps, const DSA_SIG *sig)
     *ps = sig->s;
 }
 
+int DSA_SIG_set0(BIGNUM *r, BIGNUM *s, DSA_SIG *sig)
```

Comment:
I, too, wondered at that inconsistency; the getters/setters for _e.g._ `DH` and `RSA` have the object pointer as the first argument, not the last.

However, I wanted this PR to focus just on adding the `DSA_SIG` and `ECDSA_SIG` setters; once merged, I'll create a second PR for changing the order of the arguments for getters and setters of them, for consistency.